src/zenml/integrations/label_studio/annotators/label_studio_annotator.py

```diff
@@ -0,0 +1,661 @@
+        ls = self._get_client()
+        dataset = ls.get_project(dataset_id)
+        if dataset:
+            return cast(Dict[str, Any], dataset.parsed_label_config)
+        else:
+            raise ValueError("No dataset found for the given id.")
+
+    def connect_and_sync_external_storage(
+        self,
+        uri: str,
+        config: LabelStudioDatasetSyncConfig,
+        dataset: Project,
+    ) -> Optional[Dict[str, Any]]:
+        """Syncs the external storage for the given project.
+
+        Args:
+            uri: URI of the storage source.
+            config: Configuration for the dataset.
+            dataset: Label Studio dataset.
+
+        Returns:
+            A dictionary containing the sync result.
+
+        Raises:
+            ValueError: If the storage type is not supported.
+        """
+        if self._storage_source_already_exists(uri, config, dataset):
+            return None
+        if config.storage_type == "azure":
+            if not config.azure_account_name or not config.azure_account_key:
+                logger.warn(
+                    "Authentication credentials for Azure aren't fully "
+                    "provided. Please update the storage synchronization "
+                    "settings in the Label Studio web UI as per your needs."
+                )
+            storage = dataset.connect_azure_import_storage(
+                container=uri,
+                prefix=config.prefix,
+                regex_filter=config.regex_filter,
+                use_blob_urls=config.use_blob_urls,
+                presign=config.presign,
+                presign_ttl=config.presign_ttl,
+                title=dataset.get_params()["title"],
+                description=config.description,
+                account_name=config.azure_account_name,
+                account_key=config.azure_account_key,
+            )
+        elif config.storage_type == "gcs":
+            if not config.google_application_credentials:
+                logger.warn(
+                    "Authentication credentials for Google Cloud Storage "
+                    "aren't fully provided. Please update the storage "
+                    "synchronization settings in the Label Studio web UI as "
+                    "per your needs."
+                )
+            storage = dataset.connect_google_import_storage(
+                bucket=uri,
+                prefix=config.prefix,
+                regex_filter=config.regex_filter,
+                use_blob_urls=config.use_blob_urls,
+                presign=config.presign,
+                presign_ttl=config.presign_ttl,
+                title=dataset.get_params()["title"],
+                description=config.description,
+                google_application_credentials=config.google_application_credentials,
+            )
+        elif config.storage_type == "s3":
+            if not config.aws_access_key_id or not config.aws_secret_access_key:
+                logger.warn(
```

Comment:
I think `logger.warn` is deprecated and `logger.warning` should be used instead. (same above)

Reply:
Done.